### PR TITLE
refactor: refactor command deployment & command handling

### PR DESCRIPTION
**Pull request by adrifcastr**
**Please describe the changes this PR makes and why it should be merged:**
This PR refactors parts of the interactions and command handler sections:

- command handling uses builders to ease the deployment steps
- deployment has been refactored to use @discordjs/rest entirely and not rely on other methods
- command permissions no longer showcase the messageCreate event

### CODE REVIEW

**Comment by DaStormer** [comment]
Is this a good idea? I feel like the guide should reflect the way to do this with discord.js as a “standalone.” 

These proposed changes may be misleading to new users as they may imply that the used packages are required, not just optional. Maybe a disclaimer or something that clearly shows it’s optional??

Idk, just my two cents.

**Comment by adrifcastr** [comment]
> Is this a good idea? I feel like the guide should reflect the way to do this with discord.js as a “standalone.” 

It was discussed internally that this is the approach to go for

**Comment by iCrawl** [comment]
Registering your commands should be a "separate" process.

How else are you going to do it? Let's say you don't have any slash commands, and no message intent.
Through the ready event? We cover extensively already that this is a bad idea.

So yes, having a deploy script that does all of this, great 👍

**Comment by Costpap** [comment]
> Registering your commands should be a "separate" process.
> 
> How else are you going to do it? Let's say you don't have any slash commands, and no message intent.
> Through the ready event? We cover extensively already that this is a bad idea.
> 
> So yes, having a deploy script that does all of this, great 👍

Having a deploy script that does all of this is great, except for one concern. The script assumes that `client` is available in its context. For most of the users the guide caters to, this is notoriously difficult without them extensively modify their `index.js` file as well as the script. While I'm aware that making sure one's initiated client is available as context is something that could be considered out of scope for the guide, it'd be nice to see it covered. If it's not covered, I feel like as if it'll force users to complain about their code not working in various ways, which just creates an additional burden for everyone. Or, users will just manually register their commands by sending requests to the API on their own, if they know how to do so.

**Review by Danktuary — CHANGES_REQUESTED** [review]
Looks good overall 🎉 The page order and commands registration make better sense now

File: guide/interactions/registering-slash-commands.md [hunk 1]
```diff
@@ -3,115 +3,123 @@
...
-First up, we'll introduce you to global application commands. These types of commands will be available in all guilds your application has the `applications.commands` scope authorized, as well as in DMs.
+First up, we'll introduce you to guild application commands. These types of commands are only available in the guild they were created in, if your application has the `applications.commands` scope authorized.
 
-::: tip
-Global commands are cached for one hour. New global commands will fan out slowly across all guilds and will only be guaranteed to be updated after an hour. Guild commands update instantly. As such, we recommend you use guild-based commands during development and publish them to global commands when they're ready for public use.
-:::
+In this section, we'll be using a script that is usable in conjunction with the slash command handler from the command handling section.
 
-To register a global command, pass an `ApplicationCommandData` object to the `ApplicationCommandManager#create()` method:
+First off, install the discord.js REST module by running `npm install @discordjs/rest` in your terminal.
```
Comment (Danktuary) [inline guide/interactions/registering-slash-commands.md:17]:
```suggestion
First off, install the [discord.js REST module](https://github.com/discordjs/discord.js-modules/blob/main/packages/rest/) by running `npm install @discordjs/rest` in your terminal.
```

File: guide/interactions/registering-slash-commands.md [hunk 2]
```diff
@@ -3,115 +3,123 @@
...
-## Global commands
+## Guild commands
 
-First up, we'll introduce you to global application commands. These types of commands will be available in all guilds your application has the `applications.commands` scope authorized, as well as in DMs.
+First up, we'll introduce you to guild application commands. These types of commands are only available in the guild they were created in, if your application has the `applications.commands` scope authorized.
 
-::: tip
-Global commands are cached for one hour. New global commands will fan out slowly across all guilds and will only be guaranteed to be updated after an hour. Guild commands update instantly. As such, we recommend you use guild-based commands during development and publish them to global commands when they're ready for public use.
-:::
+In this section, we'll be using a script that is usable in conjunction with the slash command handler from the command handling section.
```
Comment (Danktuary) [inline guide/interactions/registering-slash-commands.md:15]:
```suggestion
In this section, we'll be using a script that is usable in conjunction with the slash command handler from the [command handling](/command-handling/) section.
```

File: guide/interactions/registering-slash-commands.md [hunk 3]
```diff
@@ -3,115 +3,103 @@
...
-			},
-			{
-				name: 'pong',
-				description: 'Replies with Ping!',
-			},
-		];
-
-		const commands = await client.application?.commands.set(data);
-		console.log(commands);
 	}
```
Comment (Danktuary) [inline guide/interactions/registering-slash-commands.md:76]:
```suggestion
```

File: guide/interactions/registering-slash-commands.md [hunk 4]
```diff
@@ -3,115 +3,103 @@
...
+	{ body: commands },
+);
 
-## Bulk-update commands
 
-If, for example, you deploy your application commands when starting your application, you may want to update all commands and their changes at once. You can do this by passing an array of `ApplicationCommandData` objects to the `set()` method on either of the managers introduced above: 
 
-::: danger
-This will overwrite all existing commands on the application or guild with the new data provided!
-:::
```
Comment (Danktuary) [inline guide/interactions/registering-slash-commands.md:75]:
```suggestion
```

File: guide/interactions/registering-slash-commands.md [hunk 5]
```diff
@@ -3,115 +3,103 @@
...
-```
+```js {2}
+await rest.put(
+	Routes.applicationCommands(CLIENT_ID),
+	{ body: commands },
+);
 
-## Bulk-update commands
 
-If, for example, you deploy your application commands when starting your application, you may want to update all commands and their changes at once. You can do this by passing an array of `ApplicationCommandData` objects to the `set()` method on either of the managers introduced above:
```
Comment (Danktuary) [inline guide/interactions/registering-slash-commands.md:74]:
```suggestion
```

File: guide/interactions/registering-slash-commands.md [hunk 6]
```diff
@@ -3,115 +3,103 @@
...
-	}
-});
-```
+```js {2}
+await rest.put(
+	Routes.applicationCommands(CLIENT_ID),
+	{ body: commands },
+);
 
-## Bulk-update commands
```
Comment (Danktuary) [inline guide/interactions/registering-slash-commands.md:73]:
```suggestion
```

File: guide/interactions/registering-slash-commands.md [hunk 7]
```diff
@@ -3,115 +3,103 @@
...
-		const command = await client.guilds.cache.get('123456789012345678')?.commands.create(data);
-		console.log(command);
-	}
-});
-```
+```js {2}
+await rest.put(
+	Routes.applicationCommands(CLIENT_ID),
+	{ body: commands },
+);
```
Comment (Danktuary) [inline guide/interactions/registering-slash-commands.md:72]:
```suggestion
```

**Review by Danktuary — COMMENTED** [review]
Leftover lines from some of the last edits 😅

**Review by Danktuary — APPROVED** [review]

File: guide/interactions/registering-slash-commands.md [hunk 8]
```diff
@@ -3,115 +3,98 @@
 Discord provides developers with the option to create client-integrated slash commands. In this section, we'll cover how to register these commands using discord.js!
 
 ::: tip
+This page assumes you use the same file structure as our [command handling](/command-handling) section. The scripts provided are made to function with that setup.
+
 If you already have slash commands set-up for your application and want to learn how to respond to them, refer to [the following page](/interactions/replying-to-slash-commands.md).
```
Comment (muhimur9049) [inline guide/interactions/registering-slash-commands.md:8]:
```suggestion
If you already have slash commands set up for your application and want to learn how to respond to them, refer to [the following page](/interactions/replying-to-slash-commands.md).
```
`set-up` doesn't seem to fit in this context

**Pull request merged**